guides/hack/11-built-in-types/16-string.md

```diff
@@ -110,3 +110,9 @@ HHVM currently supports UTF-8, and single-byte encodings that are supported by t
 Other encodings may be supported in the future, however `Locale\set_native()` is likely to be restricted to
 the current locales; for example, UTF-16 can not be supported by `Locale\set_native()`, as UTF-16 strings can
 contain null bytes.
+
+# Working with `Regex\`
+Similar to `Str\`, functions in the `Regex\` namespace operate on bytes, not characters, unless `/u` is present in the pattern.
```

Comment:
Not all functions in `Str\` operate on bytes. Many functions in that namespace have a locale aware version f.e. `Str\length_l()` which returns the length in characters instead of bytes.

Reply:
Good point. When I wrote this I was aware of the locale versions but wanted to say something like, "Similar to the default behavior of `Str\`", but maybe it's not a fair statement to say that is the default. At the very least reading this line on its own conveys a "this is always the truth" stance, which isn't true per the `_l()` variants.

Reply:
Do you think this is clear, if the preceding part is removed? Wondering if it could be more precise.

> Functions in the `Regex\` namespace operate on bytes, not characters, unless `/u` is present in the pattern.

Reply:
It is clear, but there are ways to be more precise. I don't know if this is an improvement though. (Flags are not always preceded by a `/`, since custom pattern delimiters like `#[ab]#u` are supported.)

Functions in the `Regex\` namespace operate on bytes (ascii strings). If the string being inspected is Utf8, you should use a `Regex\Pattern` with the `u` flag. Failing to do so may result in one multi-byte character being interpreted as multiple characters.